microsoft/testsuites/vm_extensions/runtime_extensions/vmaccess.py

```diff
@@ -77,9 +80,16 @@ def _validate_password(
     node: Node, username: str, password: str, valid: bool = True
 ) -> None:
     message = f"Password not set as intended for user {username}."
+
+    if isinstance(node.os, CBLMariner):
+        if node.os.information.version >= "2.0.0":
+            # In Mariner 2.0, there is a security restriction that only allows wheel group users to use 'su' command
+            # Add current user (specified during VM creation) to wheel group in Mariner
+            node.tools[Usermod].add_user_to_group("wheel", sudo=True)
+
     # simple command to determine if username password combination is valid/invalid
     node.execute(
-        cmd=f'echo "{password}" | su --command true - {username}',
+        cmd=f'echo "{password}" | su --command true {username}',
```

Comment:
Does it need to remove the `-`?

Reply:
Please call `add_secret(password)` to add the password into secrets, so it won't appear in the log.